worldedit-core/src/main/java/com/sk89q/worldedit/command/tool/brush/ImageBrush.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * WorldEdit, a Minecraft world manipulation toolkit
+ * Copyright (C) sk89q <http://www.sk89q.com>
+ * Copyright (C) WorldEdit team and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.sk89q.worldedit.command.tool.brush;
+
+import com.sk89q.worldedit.EditSession;
+import com.sk89q.worldedit.MaxChangedBlocksException;
+import com.sk89q.worldedit.function.pattern.Pattern;
+import com.sk89q.worldedit.math.BlockVector3;
+import com.sk89q.worldedit.world.block.BaseBlock;
+import com.sk89q.worldedit.world.block.BlockTypes;
+
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.util.Random;
+
+public class ImageBrush implements Brush {
+
+    private static final Random RANDOM = new Random();
+    private final BufferedImage image;
+    private final double intensity;
+    private final boolean fill;
+    private final boolean flat;
+
+    public ImageBrush(BufferedImage image, double intensity, boolean fill, boolean flat) {
+        this.image = image;
+        this.intensity = intensity;
+        this.fill = fill;
+        this.flat = flat;
+    }
+
+    @Override
+    public void build(EditSession editSession, BlockVector3 position, Pattern pattern, double doubleSize) throws MaxChangedBlocksException {
+        // Resize the image
+        int size = (int) Math.ceil(doubleSize);
+        int diameter = size * 2 + 1;
+        BufferedImage resizedPattern = new BufferedImage(diameter, diameter, 1);
+        Graphics2D graphic = null;
+        try {
+            graphic = resizedPattern.createGraphics();
+            graphic.drawImage(this.image, 0, 0, diameter, diameter, null);
+        } finally {
+            graphic.dispose();
+        }
+
+        double random = RANDOM.nextDouble();
```

Comment:
This random feature might be a good candidate for another `@Switch`, I can see it being annoying for making exact replicas.